### PR TITLE
feat(wokwi): Add support for specifying diagram path

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
This PR adds a support of specifying diagram file for pytest-embedded-wokwi. 
To specify diagram.json add --wokwi-diagram to the pytest command

### CODE REVIEW

**Comment by hfudev** [comment]
Hi @P-R-O-C-H-Y!

The code changes LGTM. However, if I've downloaded the old wokwi-cli release and run the test, the new cli args `--wokwi-diagram` will cause an exception

```
.2024-05-30 10:26:43 _ArgError: unknown or unexpected option: --diagram-file
2024-05-30 10:26:43     at arg2 (/snapshot/dist/cli.cjs:126:23)
2024-05-30 10:26:43     at main (/snapshot/dist/cli.cjs:14099:39)
2024-05-30 10:26:43     at Object.<anonymous> (/snapshot/dist/cli.cjs:14317:1)
2024-05-30 10:26:43     at Module._compile (pkg/prelude/bootstrap.js:1926:22)
2024-05-30 10:26:43     at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
2024-05-30 10:26:43     at Module.load (node:internal/modules/cjs/loader:988:32)
2024-05-30 10:26:43     at Module._load (node:internal/modules/cjs/loader:834:12)
2024-05-30 10:26:43     at Function.runMain (pkg/prelude/bootstrap.js:1979:12)
2024-05-30 10:26:43     at node:internal/main/run_main_module:17:47 {
2024-05-30 10:26:43   code: 'ARG_UNKNOWN_OPTION'
2024-05-30 10:26:43 }
```

Could you add more code to handle this use case? I can come up with a few solutions here.

- write backward compatible code (which is quite uncessary here, since wokwi-cli is still in dev-phase)
- add one global WOKWI_MINIMUM_VERSION, and ask the user to update the wokwi cli version if the version is lower than this value.
- check and download the latest version of wokwi-cli while setup automatically (cause more network traffic to users, and maybe they're using hotspot or something)

Overall I'll pick solution 2, and welcome any other ideas!

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py [hunk 1]
```diff
@@ -51,15 +55,38 @@ def __init__(
         self.app = app
         self.firmware_resolver = firmware_resolver
 
+        # first need to check if wokwi-cli exists in PATH
+        if shutil.which('wokwi-cli') is None:
+            print('Please install wokwi-cli, by running: curl -L https://wokwi.com/ci/install.sh | sh')
+            # self.install_wokwi() not implemented yet
```
Comment (hfudev) [inline pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:61]:
```suggestion
            raise RuntimeError('Please install wokwi-cli, by running: curl -L https://wokwi.com/ci/install.sh | sh')
```

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py [hunk 2]
```diff
@@ -51,15 +55,38 @@ def __init__(
...
+            wokwi_cli_version = child.match.group(1).decode('utf-8')
+        except pexpect.TIMEOUT:
+            raise RuntimeError('Failed to run wokwi-cli')
+
+        if Version(wokwi_cli_version) < Version(WOKWI_CLI_MINIMUM_VERSION):
+            raise ValueError(
+                f'Wokwi CLI version {wokwi_cli_version} is not supported. '
+                f'Minimum version required: {WOKWI_CLI_MINIMUM_VERSION}. '
+                f'To update Wokwi CLI run: curl -L https://wokwi.com/ci/install.sh | sh'
+            )
```
Comment (hfudev) [inline pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:75]:
```suggestion
        try:
            child.expect(r'Wokwi CLI v(\d+\.\d+\.\d+)', timeout=1)
            wokwi_cli_version = child.match.group(1).decode('utf-8')
        except pexpect.TIMEOUT:
            logging.warning('Failed to get wokwi-cli version, assume version requirements satisfied')
        else:
            if Version(wokwi_cli_version) < Version(WOKWI_CLI_MINIMUM_VERSION):
                raise ValueError(
                    f'Wokwi CLI version {wokwi_cli_version} is not supported. '
                    f'Minimum version required: {WOKWI_CLI_MINIMUM_VERSION}. '
                    f'To update Wokwi CLI run: curl -L https://wokwi.com/ci/install.sh | sh'
                )
```

by this approach, if `wokwi-cli --help` changes format one day, at least we don't block the users.

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py [hunk 3]
```diff
@@ -70,6 +97,9 @@ def __init__(
     def wokwi_cli_executable(self):
         return self.WOKWI_CLI_PATH
 
+    def install_wokwi():
+        pass
+
```
Comment (hfudev) [inline pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:102]: please remove the unused functions

**Review by hfudev — CHANGES_REQUESTED** [review]

**Comment by P-R-O-C-H-Y** [comment]
@hfudev All required changes are done :)

**Review by hfudev — APPROVED** [review]
LGTM! Thanks for your PR!

**Pull request merged**